### PR TITLE
Implement fmt.Stringer interface for StreamType

### PR DESCRIPTION
**Pull request by svrana**
Following the example of `Code`, implement `fmt.Stringer` for `StreamType`. This makes it easier to use stream types in logs, traces, metric tags, and the like.

### CODE REVIEW

**Comment by svrana** [comment]
I started porting/writing a logging interceptor at github.com/svrana/go-connect-middleware and it'd be nice to be able output the StreamType as a string as is done in the grpc version of this code. Course I could do that in my own project, but thought it might generally be useful.

**Comment by bufdev** [comment]
I'm not sure we want to do this for this single enum - we'll discuss if we want to expose `fmt.Stringer` across enums. It adds to the API surface, so we want to make sure this is something we want to do.

File: connect.go [hunk 1]
```diff
@@ -53,6 +54,20 @@ const (
...
+	case StreamTypeUnary:
+		return "unary"
+	case StreamTypeClient:
+		return "client"
+	case StreamTypeServer:
+		return "server"
+	case StreamTypeBidi:
+		return "bidi"
+	}
+	return fmt.Sprintf("stream_%d", s)
```
Comment (bufdev) [inline connect.go:68]: If we do do this, generally we've just done `strconv.Itoa(int(s))` for these cases

**Comment by akshayjshah** [comment]
@svrana I'm going to reopen this, tack on a few commits, and get this merged in :)

**Review by akshayjshah — APPROVED** [review]
This looks good to me. Thanks for following the example of `Code` so closely, @svrana!

**Pull request merged**